CMakeLists.txt

```diff
@@ -372,6 +377,10 @@ if (NOT $ENV{S2N_LIBCRYPTO} STREQUAL "awslc")
     target_compile_options(${PROJECT_NAME} PRIVATE -Wcast-qual)
 endif()
 
+if (S2N_STRICT_CHECK)
+    target_compile_options(${PROJECT_NAME} PUBLIC -DS2N_STRICT_CHECK)
+endif()
```

Comment:
I'm not very familiar with cmake-- do we need both a PRIVATE and PUBLIC version of this? And why not set both at once if we do?

Reply:
I'm not familiar either, but `PRIVATE` and `PUBLIC` sets the scope of the argument following it. `PUBLIC` is required for us to use the option in`s2n_codebuild.sh`. Is my understanding correct @dougch ? Not sure if we can set both at once ( based on understanding from going through CMAKELISTS.txt)

Reply:
What happens if we only have PRIVATE? Also doesn't PUBLIC imply PRIVATE? Can you play around with these keywords and figure out if we need both?